docs/cog_guides/alias.rst

```diff
@@ -0,0 +1,270 @@
+.. _alias:
+
+=====
+Alias
+=====
+
+This is the cog guide for the alias cog. You will
+find detailed docs about the usage and the commands.
+
+``[p]`` is considered as your prefix.
+
+.. note:: To use this cog, load it by typing this::
+
+        [p]load alias
+
+.. _alias-usage:
+
+-----
+Usage
+-----
+
+This cog allows to create shortcuts for commands.
+
+You can do things like this:
+
+.. code-block:: none
+
+    [p]cleanup messages 42
+    equals to
+    [p]clean 42
+
+or even this:
+
+.. code-block:: none
+
+    [p]search sc Au temps en emporte le Z event
+    equals to
+    [p]soundcloud Au temps en emporte le Z event
+
+In the first example, we made an alias named ``clean`` that will
+invoke the ``cleanup messages`` command.
+
+In the second example, we made an alias called ``soundcloud`` that will
+invoke the ``search sc`` command (search a song to play on SoundCloud. As you
+can see, you can also add arguments to your alias. Then you can add the
+arguments you want after your alias.
+
+This is the basic usage, where you can define an alias for the first part of
+the command and give the second part when invoking the command. A more advanced
+usage of aliases is the usage of arguments.
+
+Let's suppose you want to make an alias to ban someone, delete 7 days of
+messages and set the reason to "Spam bot.", that cannot be done with a classic
+alias since the required member argument is the first one. If you create the
+alias "spamban" using arguments like this ``ban {0} 7 Spam bot.``, ``{0}`` will
+be considered as the first argument of your alias:
+
+.. code-block:: none
+
+    [p]ban Slime#3160 7 Spam bot.
+    equals to
+    [p]spamban Slime#3160
+
+For a more detailed explaination, read :ref:`this <alias-command-alias-add>`.
+
+.. _alias-commands:
+
+--------
+Commands
+--------
+
+.. _alias-command-alias:
+
+^^^^^
+alias
+^^^^^
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]alias
+
+**Description**
+
+This is the main command used for setting up the cog.
+It will be used for all other commands.
+
+.. _alias-command-alias-add:
+
+"""""""""
+alias add
+"""""""""
+
+.. note:: |mod-lock|
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]alias add <alias_name> <command>
+
+**Description**
+
+Creates an alias. It will be used like this ``[p]alias_name <arguments>``
+and will be equal to this ``[p]command <arguments>``.
+
+Let's develop a bit more the examples given :ref:`earlier <alias-usage>`,
+the left part of the command is the alias (blue), and the right part (orange)
```

Comment:
Either put the color at the end or put the color right after locating it, don't do both.